src/main/java/uk/ac/ebi/ampt2d/commons/accession/persistence/BasicSpringDataRepositoryDatabaseService.java

```diff
@@ -89,4 +96,38 @@ public void enableAccessions(List<AccessionWrapper<MODEL, String, ACCESSION>> ac
                 .collect(Collectors.toSet()));
     }
 
+    @Override
+    public AccessionWrapper<MODEL, String, ACCESSION> update(AccessionWrapper<MODEL, String, ACCESSION> accession)
+            throws AccessionDoesNotExistException, HashAlreadyExistsException {
+        Collection<ENTITY> accessionedElements = repository.findByAccessionIn(Arrays.asList(accession.getAccession()));
```

Comment:
can we use respository.findByAccession(accession.getAccession())

Reply:
The method doesn't exist right now, but it doesn't hurt to add it.